packages/utils/src/misc.ts

```diff
@@ -560,3 +560,31 @@ export function fromBase64(encodedString: string): string {
 export function sleep(ms: number) {
   return new Promise((resolve) => setTimeout(resolve, ms));
 }
+
+export class LimitedSet<T> {
+  private limit: number;
+  private set: Set<T>;
+
+  constructor({ limit }: { limit: number }) {
+    this.limit = limit;
+    this.set = new Set<T>();
+  }
+
+  add(item: T) {
+    if (this.set.has(item)) return;
+
+    if (this.set.size >= this.limit) {
```

Comment:
You're doing `>` here, so why not `while` as well? Hopefully only `==` should pass anyway, but still :D

Reply:
probably `while` is an overkill for what it does currently